wordpress-mobile/WordPress-iOS
# Add new feature flag for lock screen widgets in Jetpack

Related to: https://github.com/wordpress-mobile/WordPress-iOS/issues/19411

This is the first PR for adding the stats widget to the lock screen
1.  https://github.com/wordpress-mobile/WordPress-iOS/pull/20309 (✅ Approved) 👈 you're here!
- Update compilation flag to `AppConfiguration.isJetpack`
2. https://github.com/wordpress-mobile/WordPress-iOS/pull/20312 (✅ Approved)
3. https://github.com/wordpress-mobile/WordPress-iOS/pull/20342 (TBD)
4. https://github.com/wordpress-mobile/WordPress-iOS/pull/20353 (✅ Approved)
5. https://github.com/wordpress-mobile/WordPress-iOS/pull/20371 (✅ Approved)
6. https://github.com/wordpress-mobile/WordPress-iOS/pull/20317 (✅ Approved)
7. https://github.com/wordpress-mobile/WordPress-iOS/pull/20368 (✅ Approved)
8. https://github.com/wordpress-mobile/WordPress-iOS/pull/20399 (✅ Approved)
9. https://github.com/wordpress-mobile/WordPress-iOS/pull/20405 (✅ Approved)
(Confirm the data display on UI correctly in this phase)
10. https://github.com/wordpress-mobile/WordPress-iOS/pull/20422 (✅ Approved)
11. https://github.com/wordpress-mobile/WordPress-iOS/pull/20427 (In Reviewing)

## Description
1. Add a new feature flag `lockScreenWidget` to protect the following change from being released before it is verified.
Set the `canOverride` to `false` because the widget list updates along with the app installation. 
Dynamic changes of the flag do not work.

2. Add ~~`JETPACK_STATS_WIDGETS` compilation condition~~ **Update to AppConfiguration.isJetpack** to allow the lock screen widget to be added only for the Jetpack target.

3. Add a new widget kind "WordPressLockScreenWidgetTodayViews" for the first style widget 

4. Add a new empty widget for testing

## Images & Videos
| Jetpack in add widgets | New widget in selection page |
| - | - |
|![JetpackInLockScreen](https://user-images.githubusercontent.com/3096210/224618653-7d2f68da-668e-4096-8d96-9133382aa543.png)|![JetpackWidgetSelection](https://user-images.githubusercontent.com/3096210/224618628-525019bf-fcd9-46e3-9ec1-499e61be49c9.png)|

## Testing instructions
- When the feature flag on, should be able to see Jetpack and add the new widget to lock screen.
- When the feature flag on, shouldn't see the WordPress in widget selection
- When the feature flag off, shouldn't see the Jetpack and WordPress in widget selection

## Regression Notes
1. Potential unintended areas of impact
Existing widgets in WordPress/Jetpack

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Added existing home screen widgets first, it works well after setting up new widgets.
- After setting up new widgets, adding existing home screen widgets flow won't be impacted

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
